server/apps/highlights/generate.py

```diff
@@ -48,15 +44,23 @@ def create(self, docs, **kwargs):
             doc['task'] = package.get('task')
             doc['family_id'] = package.get('guid')
 
-            body = []
+            items = []
             for group in package.get('groups', []):
                 for ref in group.get('refs', []):
                     if 'residRef' in ref:
                         item = service.find_one(req=None, _id=ref.get('residRef'))
-                        body.append(item_to_str(item))
-                        body.append('<p></p>')
-            doc['body_html'] = join_html(body)
-
+                        if item:
+                            html = item.get('body_html')
+                            if html:
+                                soup = BeautifulSoup(html)
+                                item['body_html'] = str(soup.p)
```

Comment:
ok now I see it's here. maybe it would be better to put there `first_paragraph` and also the body itself so you can do something else with those templates